types/node/index.d.ts

```diff
@@ -5938,36 +5938,36 @@ declare module "crypto" {
 declare module "stream" {
     import * as events from "events";
 
-    class internal extends events.EventEmitter {
-        pipe<T extends NodeJS.WritableStream>(destination: T, options?: { end?: boolean; }): T;
+    class internal<T> extends events.EventEmitter {
```

Comment:
I've been meaning to create a PR to remove the `internal` type for awhile. Adding a generic type argument to it makes it even worse. A better approach would be:

```ts
declare module "stream" {
    import * as events from "events";
    import stream = require("stream");

    class Stream<T = any> extends events.EventEmitter {
        pipe<TDestination extends NodeJS.WritableStream<T>>(destination: TDestination, options?: { end?: Boolean; }): TDestination;
    }

    namespace Stream {
        export import Stream = stream;
        // …
    }

    export = Stream;
```